docs/architecture/pairing-with-use-integration-system.md

```diff
@@ -0,0 +1,86 @@
+
+1. After successful pairing Integration System Instance can manage the Application (7).
+
+>**NOTE** The pairing process described here allows the Application to establish trusted relation with its Integration System Instance and does not establish trusted relation between the Application and its Runtimes.
+
+## Details
+
+The Compass supports multiple Integration System Instances (represented internally by Integration Systems) to help abstract the way how Applications and Runtimes can be managed. Each Integration System can have its own approach for obtaining details for establishing a trusted relation for the managed system (Application, Runtime). To make the Director architecture clean and improve users' experience we introduce the concept of Pairing Adapters. Each Integration System can be coupled with Pairing Adapter. Thanks to this approach the Directors' API can leverage different Integration Systems under single mutation and handle the requests in a consistent manner. When Administrator wants to establish a new trusted relation between Application and Integration System the Director API offers `generateAuthForApplication(id: ID!, type: PairingAdapterAuthTypeInput!, inputParams: JSON!)` mutation.
+
+Mutations `generateAuthForApplication(id: ID!, type: PairingAdapterAuthTypeInput!, inputParams: JSON!)` abstracts the way of obtaining trusted relation data from the Integration System. Calling the mutation with appropriate parameters checks whether the desired Application is under control of the Integrations System. If the underlying Integration System has Pairing Adapter assigned, the call is enriched with additional data defined for Integration System and data which comes from the request context (eg. user ID). Next, the request is handed to the Pairing Adapter. The Pairing Adapter implements the logic required for obtaining all required data from the Integration System Instance based on the incoming request.
+
+In case the Application is not related to any Integration System the `generateAuthForApplication(id: ID!, type: PairingAdapterAuthTypeInput!, inputParams: JSON!)` mutation abstracts fetching the one-time token from the Connector or client credentials, according to the credential type described by the `type` parameter.
+
+The Integration System can support multiple auth types. To determine what are the supported types for a particular Application, the API offers appropriate query: `getAuthTypesForApplication(id: ID!)`. The response for the query returns the possible values for the `type` parameter for auth generation mutation.
+
+## Pairing Adapter
+
+Pairing Adapter is a component that runs in the Compass environment. Its API is not exposed outside the cluster. It is responsible for handling specific logic to obtain auth details required to establish a trusted relation between the Application Instance and Integration System Instance. Each Pairing Adapter is coupled with one Integration System.
```

Comment:
we should highlight that Pairing Adapter is deployed as a service, in the same cluster as Compass. So it also means, that deploying such adapter has to be done by the cluster administrator. Have you consulted this idea with Security Experts?

Reply:
> we should highlight that Pairing Adapter is deployed as a service, in the same cluster as Compass

And, additionally, please highlight that the communication between Compass and Pairing Adapter is always internal, and, in a result, is trusted because of mTLS